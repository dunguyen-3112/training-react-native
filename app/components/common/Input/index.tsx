import { StyleSheet, TextInput, View } from 'react-native';
import React, { useMemo, useCallback, memo } from 'react';
import Text from '../Text';
import { Font, COLOR } from '@constants';

interface InputProps {
  field: string;
  value: string;
  label?: string;
  labelFont?: Font;
  labelColor?: COLOR;
  inputFont?: Font;
  inputColor?: COLOR;
  placeholder?: string;
  placeholderColor?: string;
  error?: string;
  errorColor?: string;
  onChangeText: (value: string, field?: string) => void;
  onBlur?: () => void;
  onFocus?: () => void;
}
const Input = ({
  field,
  value,
  label,
  labelFont,
  labelColor,
  inputFont,
  inputColor,
  placeholder,
  onChangeText,
  onBlur,
  onFocus,
}: InputProps) => {
  const size = inputFont?.fontSize;

  const _size = useMemo(() => {
    if (typeof size === 'number') return size;
    switch (size) {
      case 'large':
        return 24;
      case 'medium':
        return 14;
      case 'small':
        return 10;
      default:
        return 12;
    }
  }, [size]);

  const handleChangeInput = useCallback(
    (value: string) => {
      onChangeText(value, field);
    },
    [field, onChangeText]
  );

  return (
    <View>
      {label && (
        <Text color={labelColor || 'black'} font={labelFont}>
          {label}
        </Text>
      )}
      <TextInput
        value={value}
        placeholder={placeholder}
        onBlur={onBlur}
        onFocus={onFocus}
        onChangeText={handleChangeInput}
        style={[
          inputFont
            ? {
                color: inputColor,
                fontSize: _size,
                fontWeight: inputFont?.fontWeight || '400',
              }
            : styles.input,
        ]}
      />
    </View>
  );
};

export default memo(Input);

const styles = StyleSheet.create({
  input: {
    fontSize: 13,
    fontWeight: '400',
    color: '#DA6317',
    paddingVertical: 14,
    paddingHorizontal: 60,
  },
});
