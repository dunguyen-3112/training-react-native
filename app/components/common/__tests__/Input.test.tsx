import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input';

const fakeInput = {
  field: 'Name',
  value: 'User Name',
  onChangeText: jest.fn(),
};

describe('Input', () => {
  const input = renderer.create(<Input {...fakeInput} />);
  it('renders correctly', () => {
    expect(input).toMatchSnapshot();
  });
});
