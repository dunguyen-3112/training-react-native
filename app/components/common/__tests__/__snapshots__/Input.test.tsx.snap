// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<View>
  <TextInput
    onChangeText={[Function]}
    placeholderTextColor="#000000"
    style={
      Array [
        Object {
          "color": "#DA6317",
          "fontSize": 13,
          "fontWeight": "400",
          "paddingHorizontal": 60,
          "paddingVertical": 14,
        },
        Object {
          "color": "#000000",
        },
      ]
    }
    value="User Name"
  />
</View>
`;
